app/src/main/java/com/internship/firstweekapp/ui/notes_list/NotesListFragmentViewModel.kt

```diff
@@ -0,0 +1,100 @@
+package com.internship.firstweekapp.ui.notes_list
+
+import androidx.databinding.ObservableArrayList
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.viewModelScope
+import com.internship.firstweekapp.arch.BaseViewModel
+import com.internship.firstweekapp.arch.lifecycle.SingleLiveEvent
+import com.internship.firstweekapp.room.DatabaseProvider
+import com.internship.firstweekapp.room.Note
+import com.internship.firstweekapp.ui.notes_list.note_item.NoteViewModel
+import com.internship.firstweekapp.util.Direction
+import com.internship.firstweekapp.util.SortBy
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class NotesListFragmentViewModel(val databaseProvider: DatabaseProvider) : BaseViewModel() {
+
+    val list = ObservableArrayList<NoteViewModel>()
+    val sortBy = MutableLiveData<SortBy>()
+    private val sortTypeObserver = Observer<SortBy> {
+        viewModelScope.launch(Dispatchers.IO) {}
+    }
+
+    val direction = MutableLiveData<Direction>()
+
+    val navigationEvent = SingleLiveEvent<Boolean>()
+
+    private val directionObserver = Observer<Direction> {
+        viewModelScope.launch(Dispatchers.IO) {
+            list.clear()
+            setList(it)
+        }
+    }
+
+    init {
+        direction.observeForever(directionObserver)
+        sortBy.observeForever(sortTypeObserver)
+        sortBy.postValue(SortBy.Title)
+        direction.postValue(Direction.Reverse)
+    }
+
+    fun setList(it: Direction) {
```

Comment:
sorting shoulb be part of sql query - just pass sort type and sort direction into sql - then getting data and show it in recycleview